### PR TITLE
AArch64: Implement tableEvaluator

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements tableEvaluator() for AArch64.

Co-authored-by: KONNO Kazuhiro konno@jp.ibm.com
Co-authored-by: Aaron Graham aaron.graham@unb.ca
Co-authored-by: Md. Mahbubur Rahman mrahma15@unb.ca

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
This PR is based on the code from #4214.

It generates the following instruction sequence for the `5 <= numBranchTableEntries` case.
```
b.cs Label for default child
adr tmpReg, #12 // tmpReg points to the first instruction of the jump table
add tmpReg, tmpReg, selectorReg, lsl #2 // tmpReg = selectorReg*4 + tmpReg
br tmpReg
b Label Lxxxx
b Label Lxxxx
b Label Lxxxx
```

File: compiler/aarch64/codegen/ControlFlowEvaluator.cpp [hunk 1]
```diff
@@ -514,8 +515,70 @@ OMR::ARM64::TreeEvaluator::lookupEvaluator(TR::Node *node, TR::CodeGenerator *cg
...
+         {
+         generateCompareImmInstruction(cg, node, selectorReg, i);
+         generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, node->getChild(2+i)->getBranchDestination()->getNode()->getLabel(), TR::CC_EQ);
+         }
+
+      generateLabelInstruction(cg, TR::InstOpCode::b, node, defaultChild->getBranchDestination()->getNode()->getLabel(), conditions);
+      }
+   else
+      {
+      if (!constantIsUnsignedImm12(numBranchTableEntries))
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/ControlFlowEvaluator.cpp]: I think this `if` condition is being overly defensive.  You already know from the `if` guarding entry to this code that the `numBranchTableEntries` is <= 5.  I'd say just do the `generateCompareImmInstruction`.

Comment (knn-k) [reply compiler/aarch64/codegen/ControlFlowEvaluator.cpp]: `numBranchTableEntries` is >=5 here.  So it is possible that exceeds the range of unsigned imm12.

Comment (0xdaryl) [reply compiler/aarch64/codegen/ControlFlowEvaluator.cpp]: Ugh.  I read the condition above incorrectly.

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

**Pull request merged**